packages/web-components/fast-foundation/src/date-picker/date-picker.template.ts

```diff
@@ -0,0 +1,380 @@
+        ${repeat(
+            x => x.arrayToMatrix(items, 4),
+            html`
+            <${row}
+                role="row"
+                role-type="default"
+                class="picker-row"
+                part="picker-row"
+                grid-template-columns="1fr 1fr 1fr 1fr"
+            >
+            ${repeat(
+                x => x,
+                html`
+                <${cell}
+                    tabindex="-1"
+                    class="picker-cell ${x => (x.selected ? "selected" : "")}"
+                    part="picker-cell"
+                    grid-column="${(x, c) => c.index + 1}"
+                    @click="${x => x.action()}"
+                    @keyup="${(x, c) => x.keyup(c.event)}"
+                >
+                ${x => x.text}
+              </${cell}>
+            `,
+                { positioning: true }
+            )}
+          </${row}>
+        `
+        )}
+        </${grid}>
+    `;
+};
+
+/**
+ *
+ * @param context - The date-picker class context
+ * @param definition - Date-picker options
+ * @returns - A ViewTemplate
+ * @public
+ */
+export const datePickerTemplate: FoundationElementTemplate<
+    ViewTemplate<DatePicker>,
+    DatePickerOptions
+> = (context, definition) => {
+    const textField: string = context.tagFor(TextField);
+    const anchoredRegion: string = context.tagFor(AnchoredRegion);
+    const button: string = context.tagFor(Button);
+    const calendar: string = context.tagFor(Calendar);
+    const resetButton: ViewTemplate<DatePicker> = html`<${button} class="reset-text" part="reset-text" @click="${x =>
+        x.resetCalendar()}">${x => x.resetText}</${button}>`;
+
+    return html`
+    <template
+        @mouseover="${x => (x.overFlyout = true)}"
+        @mouseout="${x => (x.overFlyout = false)}"
+        @focusout="${(x, c) => x.handleFocusOut(c.event)}"
+        @focus="${x => x.handleFocus()}"
+    >
+       <${textField}
+            class="text-field"
+            part="text-field"
+            ${ref("textField")}
+            name=${x => x.name}
+            @click="${x => (!x.readonly ? x.toggleFlyout(true) : () => true)}"
+            ?readonly="${x => x.readonly}"
+            placeholder="${x => x.placeholder}"
+            appearance="${x => x.appearance}"
+            ?disabled="${x => x.disabled}"
+            ?required="${x => x.required}"
+            @blur="${x => x.handleBlur()}"
+            @keyup="${(x, c) => x.handleKeyup(c.event as KeyboardEvent)}"
+        >
+            <slot slot="start" name="start" part="start"></slot>
+            <slot></slot>
+            <slot name="end" slot="end" part="end">
+                ${
+                    definition.calendarIcon instanceof Function
+                        ? definition.calendarIcon(context, definition)
+                        : definition.calendarIcon ?? ""
+                }
+            </slot>
+        </${textField}>
+        ${when(
+            x => !x.readonly && !x.disabled,
```

Comment:
` x => !x.readonly && !x.disabled && x.flyoutOpen,`

You only want the anchored region to be added to the DOM when it is visible, otherwise it is still "doing things" in the background.  This also fixes the flicker.

Reply:
Note that the ` dateInString(date, datesString) ` function in calendar.js sometimes throws an exception related to datesString being null with this change.

Reply:
Good call on using when(). Updated.

Fixed dateInString. Added null checks for date && dateString.